src/Faction/formulas/donation.ts

```diff
@@ -1,7 +1,31 @@
+import { Person } from "@nsdefs";
+
 import { CONSTANTS } from "../../Constants";
 import { currentNodeMults } from "../../BitNode/BitNodeMultipliers";
-import { Person as IPerson } from "@nsdefs";
+import { PlayerObject } from "../../PersonObjects/Player/PlayerObject";
+import { Faction } from "../Faction";
 
-export function repFromDonation(amt: number, person: IPerson): number {
+export function repFromDonation(amt: number, person: Person): number {
   return (amt / CONSTANTS.DonateMoneyToRepDivisor) * person.mults.faction_rep * currentNodeMults.FactionWorkRepGain;
 }
+
+export function repNeededToDonate(): number {
+  return Math.floor(CONSTANTS.BaseFavorToDonate * currentNodeMults.RepToDonateToFaction);
+}
+
+export function canDonate(amt: number, player: PlayerObject): boolean {
+  return !isNaN(amt) && amt > 0 && player.money >= amt;
+}
+
+/** Donates money to the faction provided and returns repuation gained */
+export function donate(amt: number, player: PlayerObject, faction: Faction) {
+  if (!canDonate(amt, player)) {
+    return 0;
+  }
+
+  const repGain = repFromDonation(amt, player);
+  player.loseMoney(amt, "other");
+  faction.playerReputation += repGain;
+
+  return repGain;
+}
```

Comment:
For functions that use *the* player as the actor, we just use the imported player, there's no need to pass the player around as a function argument if there's exactly one valid thing you can pass as that argument. If it's allowed to be a Person in general, we would take in a Person instead of a PlayerObject.